components/SQLEditor.go

```diff
@@ -77,3 +82,95 @@ func (s *SQLEditor) SetBlur() {
 	s.SetBorderColor(tcell.ColorWhite)
 	s.SetTextStyle(tcell.StyleDefault.Foreground(tcell.ColorWhite))
 }
+
+
+/*
+	THIS FUNCTION OPENS EXTERNAL EDITOR.
+
+	Q: WHY OPEN ANOTHER TERMINAL?
+	A: OPENING EDITORS LIKE VIM/NEOVIM REALLY MESSED UP INITIAL TERMINAL'S OUTPUT.
+*/
+
+func openExternalEditor(s *SQLEditor) string {
+	// Path of temporary file
+	path := "/tmp/lazysql.sql"
+
+	editor := getEditor()
+	terminal := getTerminal()
+
+	// Create a temporary file with the current SQL query content
+	content := []byte(s.TextArea.GetText())
+
+	/*
+	0644 Permission
+	* User: read & write
+	* Group: read
+	* Other: read
+	*/
+
+	err := os.WriteFile(path, content, 0644)
```

Comment:
The syntax is octal, it should be 0o644

Reply:
I have added prefix 0, and I think that makes it a octal value.

Reply:
It does. It's valid.

The one I suggested was introduced in Go 1.13, and should be preferred if we consider some linters.

https://github.com/mvdan/gofumpt/issues/51